src/dashboard/src/components/api/views.py

```diff
@@ -890,21 +892,31 @@ def format_task(task, detailed_output=False):
     return result
 
 
+def remove_prefix(s, prefix):
+    """Remove a prefix from a string"""
+    if s.startswith(prefix):
+        return s[len(prefix) :]
+    return s
+
+
 @_api_endpoint(expected_methods=["GET"])
 def unit_jobs(request, unit_uuid):
-    jobs = models.Job.objects.filter(sipuuid=unit_uuid)
+    """Return jobs associated with a unit's UUID"""
+    jobs = models.Job.objects.filter(sipuuid=unit_uuid).order_by("createdtime")
     if not jobs.count():
         return _error_response("No jobs found for unit: {}".format(unit_uuid))
     result = []
     microservice = request.GET.get("microservice")
     if microservice is not None:
-        jobs = jobs.filter(microservicegroup=microservice)
+        microservice = remove_prefix(microservice, "Microservice:")
+        jobs = jobs.filter(microservicegroup=microservice.strip())
```

Comment:
This is great. Thanks for making this change and the one for `Job:`. 

If `remove_prefix` performs the `strip()` function too, then this can become:

    jobs = jobs.filter(microservicegroup=remove_prefix(microservice))

But I'm not sure where we all stand on code density! :)

Reply:
I thought about it, but `remove_prefix` seemed like a good and simple name for the intended use and I couldn't think of a better name if I added the `strip` call in it (`extract_name` seemed too generic, etc). And after realizing that `link_uuid` should be stripped too I decided to go with three consistent `strip` calls inside the `filter` calls instead.